hi-ml-azure/tests/testazure/test_azure_util.py

```diff
@@ -636,3 +705,114 @@ def test_get_aml_runs(tmp_path: Path) -> None:
+        source_directory=".",
+        command=["cd ."],  # command that does nothing
+        compute_target="local"
+    )
+    run = experiment.submit(config)
+
+    file_to_upload = tmp_path / "dummy_file.txt"
+    with open(file_to_upload, "w+") as f_path:
+        f_path.write("Hello world")
+
+    # This should store the file in outputs
+    run.upload_file("dummy_file", str(file_to_upload))
+
+    output_file_path = tmp_path / "downloaded_file.txt"
+    assert not output_file_path.exists()
+
+    start_time = time.perf_counter()
+    util.download_run_file(run, "dummy_file", output_file_path)
+    end_time = time.perf_counter()
+    time_dont_validate_checksum = end_time - start_time
+
+    assert output_file_path.exists()
+    with open(output_file_path, 'r') as f_path:
+        data = f_path.read()
+        assert data == "Hello world"
+
+    # Now delete the file and try again with _validate_checksum == True
+    os.remove(output_file_path)
+    assert not output_file_path.exists()
+    start_time = time.perf_counter()
+    util.download_run_file(run, "dummy_file", output_file_path, validate_checksum=True)
+    end_time = time.perf_counter()
+    time_validate_checksum = end_time - start_time
+
+    assert output_file_path.exists()
+    with open(output_file_path, 'r') as f_path:
+        data = f_path.read()
+        assert data == "Hello world"
+
+    logging.info(f"Time to download file without checksum: {time_dont_validate_checksum} vs time with"
+                 f"validation {time_validate_checksum}.")
+
+
+def _get_file_names(pref: str = "") -> List[str]:
+    file_names = ["somepath.txt", "abc/someotherpath.txt", "abc/def/anotherpath.txt"]
+    if len(pref) > 0:
+        return [u for u in file_names if pref in u]
+    else:
+        return file_names
+
+
+def test_get_run_file_names() -> None:
+    with patch("azureml.core.Run") as mock_run:
+        expected_file_names = _get_file_names()
+        mock_run.get_file_names.return_value = expected_file_names
+        # check that we get the expected run paths if no filter is applied
+        run_paths = util.get_run_file_names(mock_run)  # type: ignore
+        assert len(run_paths) == len(expected_file_names)
+        assert sorted(run_paths) == sorted(expected_file_names)
+
+        # Now check we get the expected run paths if a filter is applied
+        prefix = "abc"
+        expected_file_names_filtered = _get_file_names(pref=prefix)
+        run_paths = util.get_run_file_names(mock_run, prefix=prefix)
+        assert len(run_paths) == len(expected_file_names_filtered)
+
+
+@pytest.mark.parametrize("prefix", ["", "abc"])
+def test_download_run_files(tmp_path: Path, prefix: str) -> None:
+    # Assert that 'downloaded' paths don't exist to begin with
+    dummy_paths = [x[0] for x in _get_file_names(pref=prefix)]
+    expected_paths = [tmp_path / dummy_path for dummy_path in dummy_paths]
+    assert sum([p.exists() for p in expected_paths]) == 0
```

Comment:
sum(...)==0 is a very numpy-y way of expressing it. `assert not any(...)` is clearer, in my view.